### PR TITLE
Feature/separate ajax files

### PR DESCRIPTION
**Pull request by dingo-d**
Removed admin-ajax usage for theme sniffing, and instead use REST API.

Added endpoints for sniff initialization and for individual sniff callback files.

### CODE REVIEW

**Comment by dingo-d** [comment]
@grappler @ernilambar Can you take a look at this?

**Comment by dingo-d** [comment]
I ran the test on a clean install on VVV with test underscores theme, and I got these results:

screen shot 2017-10-24 at 08 34 38

File: js/admin.js [hunk 1]
```diff
@@ -93,6 +101,8 @@ jQuery( document ).ready(function($) {
 				file_no++;
 				if (file_no < total_files) {
 					individualSniff( theme_name, theme_args, theme_files, total_files, file_no );
+				} else {
+					$sniff_report.after('<div>' + localizationObject.check_done + '</div>');
```
[the remark this answers is not in the document]
Comment (dingo-d) [reply js/admin.js]: This is fixed in the new commit.

**Comment by grappler** [comment]
It would be interesting to compare it to the ajax version on the same system.

**Comment by dingo-d** [comment]
This is the result using AJAX: 
screen shot 2017-10-24 at 08 51 27

Not a big difference but seems a bit faster. Which is kinda odd, as REST should generally be faster.

**Comment by ernilambar** [comment]
What are the pros of using REST if performance is similar?

**Comment by dingo-d** [comment]
Well on the admin side `admin-ajax.php` is fast (admin is already loaded), but on the front it's slower since it is loading the entire admin to run it.
There is an explanation here: https://wordpress.stackexchange.com/a/273781/58895

My idea was using more modern approach plus (from what I understand) admin-ajax will be replaced by REST in the future.

**Comment by dingo-d** [comment]
@grappler @ernilambar So what's the verdict about this? :)

**Comment by ernilambar** [comment]
@dingo-d :+1: from my side.

**Comment by grappler** [comment]
I am fine with merging it. I just thought it is important that we compare the two.

File: inc/routes.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ *
+ * @since 0.1.0
+ */
+function theme_sniffer_endpoint_init() {
+	register_rest_route( 'wp/v2/theme-sniffer/', '/sniff-run', array(
+		'methods'  => 'POST',
+		'callback' => 'theme_sniffer_run_sniffer',
+	) );
+
+	register_rest_route( 'wp/v2/theme-sniffer/', '/individual-sniff', array(
```
Comment (grappler) [inline inc/routes.php]: @dingo-d As I was reading up on `register_rest_route()` I realized we should use `theme-sniffer/v1` as the namespace. Could you make a PR to change this?
https://developer.wordpress.org/rest-api/extending-the-rest-api/routes-and-endpoints/#namespaces
https://developer.wordpress.org/rest-api/extending-the-rest-api/adding-custom-endpoints/#bare-basics

Comment (dingo-d) [reply inc/routes.php]: Sure thing, PR coming right up.

**Pull request merged**